### PR TITLE
Added Heading.

### PR DESCRIPTION
**Pull request by ericsoderberghp**
The Heading differs from 1.0 in:

The `tag` property has been superseded by the `level` property.
There is a new `size` property to enable stylistic adjustment.
`margin` now supports an object allowing just top or bottom margin.
`strong` has been removed. Callers should use `<strong>`.
`uppercase` has been removed. Callers should use toUpperCase() when populating.

### CODE REVIEW

**Comment by alansouzati** [comment]
I like all the simplifications and changes. well done!

File: src/js/components/heading/Heading.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  render() {
+    const {
+      level,
+      ...rest
+    } = this.props;
+
+    const tag = `h${level}`;
+    let StyledComponent = styledComponents[tag];
+    if (!StyledComponent) {
+      StyledComponent = StyledHeading.withComponent(tag);
```
Comment (alansouzati) [inline src/js/components/heading/Heading.js]: for accessibility reasons I propose we don't allow changing the tag here.

Users with disabilities use tools to see all the heading in the page. and if we are not using the h1-h6 range they will not be able to see it.

File: src/js/components/heading/StyledHeading.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+const StyledHeading = styled.h1`
+  ${props => sizeStyle(props)}
+  ${props => props.margin && marginStyle(props)}
+  ${props => props.textAlign && textAlignStyle}
+  ${props => props.truncate && truncateStyle}
+`;
+
+export default StyledHeading.extend`
+  ${props => props.theme.box && props.theme.box.extend}
```
Comment (alansouzati) [inline src/js/components/heading/StyledHeading.js]: I believe this should be `props.theme.heading`

File: src/js/components/heading/Heading.js (hunk 1 above)
Comment (ericsoderberghp) [inline src/js/components/heading/Heading.js]: Callers can only change the level to something from 1-4. So, the tag will be "h1", "h2", "h3", or "h4" only.

File: src/js/components/heading/StyledHeading.js (hunk 2 above)
Comment (ericsoderberghp) [inline src/js/components/heading/StyledHeading.js]: Oops. Good catch.

File: src/js/components/heading/Heading.js (hunk 1 above)
Comment (alansouzati) [inline src/js/components/heading/Heading.js]: of course. I missed that. sorry

**Pull request merged**